tests/certification/pubsub/azure/eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,301 @@
+				m.Assert(ctx, 25*timeout)
+			}
+
+			return nil
+		}
+	}
+
+	// simulate the publish of messages to iot endpoint (./send-iot-device-events.sh)
+	// addExpectedMessagesforIot := func(messageWatchers *watcher.Watcher) flow.Runnable {
+	// 	return func(ctx flow.Context) error {
+	// 		messages := make([]string, numMessages)
+	// 		for i := range messages {
+	// 			messages[i] = fmt.Sprintf("testmessageForEventHubCertificationTest #%v", i+1)
+	// 		}
+	// 		messageWatchers.ExpectStrings(messages...)
+	// 		return nil
+	// 	}
+	// }
+
+	flow.New(t, "eventhubs certification").
+
+		// Test : single publisher, multiple subscriber with their own consumerID
+		// Run subscriberApplication app1
+		Step(app.Run(appID1, fmt.Sprintf(":%d", appPort),
+			subscriberApplication(appID1, topicName1, consumerGroup1))).
+
+		// Run the Dapr sidecar with the eventhubs component 1.
+		Step(sidecar.Run(sidecarName1,
+			embedded.WithComponentsPath("./components/consumer1"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// Run subscriberApplication app2
+		Step(app.Run(appID2, fmt.Sprintf(":%d", appPort+portOffset),
+			subscriberApplication(appID2, topicName1, consumerGroup2))).
+
+		// Run the Dapr sidecar with the component 2.
+		Step(sidecar.Run(sidecarName2,
+			embedded.WithComponentsPath("./components/consumer2"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+
+		// Run dummy app
+		Step(app.Run(dummyAppID, fmt.Sprintf(":%d", appPort+portOffset*10),
+			dummyApp(dummyAppID))).
+		// Run the dummy Dapr sidecar with the dummyconsumer component, to be used for publishing messages to a topic with no subscribers (unused)
+		Step(sidecar.Run(sidecarDummyName,
+			embedded.WithComponentsPath("./components/dummyconsumer"),
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort+portOffset*10),
+			embedded.WithDaprGRPCPort(runtime.DefaultDaprAPIGRPCPort+portOffset*10),
+			embedded.WithDaprHTTPPort(runtime.DefaultDaprHTTPPort+portOffset*10),
+			embedded.WithProfilePort(runtime.DefaultProfilePort+portOffset*10),
+			runtime.WithSecretStores(secretStoreComponent),
+			runtime.WithPubSubs(component))).
+		Step("publish messages to topic1", publishMessages(nil, sidecarName1, topicName1, consumerGroup1, consumerGroup2)).
+		// Start a sidecar with a component refering eh connection string of unusedTopic
+		Step("publish messages to unUsedTopic", publishMessages(nil, sidecarDummyName, unUsedTopic)).
```

Comment:
you can use sideCar1 itself to publish to a different topic ... ? Why is the dummy sidecar needed?

Reply:
eh connectionstring provided in component (`consumer1\eventhubs.yaml`) which is loaded by sideCar1 is scoped at topic level. With this constraint it won't be possible to use sidecar1 to publish to a different topic.

For reference : (EntityPath part of connection string : topic name)

`"Endpoint=sb://*.servicebus.windows.net/;SharedAccessKeyName=**;SharedAccessKey=*;EntityPath=certification-pubsub-topic1"`

This will be possible only if provide connection string from the policy defined at namespace level.

Reply:
Notes from offline conversation with Mukundan.

- Test for connection string at namespace level.
- Publishing to multiple topics with a single sidecar.
	- subscription to multiple topics with single sidecar
	- Sidecar 1 : namespace level
		- Multiple topics

Reply:
Fixed.